### PR TITLE
Feature/response controller

### PR DESCRIPTION
**Pull request by avishkar58**
Implementation of #43

### CODE REVIEW

**Comment by Carvalho96v2** [comment]
Looks amazing!!

Quick question, why do you use the mixin for the get request, but not for the post and put requests? What is the general best practice with regards to mixins? :)

File: api/app/utils/testing.py [hunk 1]
```diff
@@ -16,5 +27,4 @@ def setUp(self):
     def tearDown(self):
         db.session.remove()
         db.reflect()
-        db.engine.execute('DELETE FROM alembic_version;')
```
Comment (Carvalho96v2) [inline api/app/utils/testing.py]: Also, why did you remove this line? My understanding of this is that it "resets" the current migration?

**Comment by avishkar58** [comment]
Hey, I used the ApplicationFormMixin in the get request because it already defined a request parser with an event_id argument which I needed there. The push, post and delete didn't need event_id and I didn't create new Mixins because I'm not convinced about them :-P - it only makes sense if we're re-using a bunch of validation code in many places, which we aren't. Even if we end up doing so, it's better to follow YAGNI and refactor when that time comes...

File: api/app/utils/testing.py (hunk 1 above)
Comment (avishkar58) [reply api/app/utils/testing.py]: This was done in conjunction with switching to SQLLite for our tests. Previously we ran against the "main" postgres database, which already had migrations run against it and therefore had an alembic_version table. I believe now the tables are created directly in SQLLite with db.reflect() (@jaderabbit can correct me if I'm wrong?) - ie. no migrations therefore no alembic_version table.

Comment (Carvalho96v2) [reply api/app/utils/testing.py]: Okay cool, that makes sense :)
